blocks/v2-footer/v2-footer.js

```diff
@@ -0,0 +1,207 @@
+  let logoLink = null;
+
+  if (picture) {
+    logoLink = picture.closest('div').querySelector('p');
+    logoLink?.classList.add('footer-logo-link');
+
+    logoLink.innerHTML = '';
+    logoLink.append(picture);
+  }
+
+  const bottomLinksList = [...footer.querySelectorAll('ul')].at(-1);
+  bottomLinksList?.classList.add('v2-footer-bottom-section');
+
+  const formSection = footer.querySelector('.eloqua-form');
+  formSection?.classList.add('footer-form-section');
+
+  const footerTemplate = `
+    <div class="v2-footer-content">
+      <div class="v2-footer-main-content">
+        <div class="v2-footer-logo-section v2-logo">
+          ${logoLink?.outerHTML}
+        </div>
+        <div class="v2-footer-social-section v2-social">
+          ${socialMediaSection?.outerHTML}
+        </div>
+        <div class="v2-footer-links-section v2-content">
+          ${firstLinks?.outerHTML}
+          ${secondLinks?.outerHTML}
+          ${thirdLinks?.outerHTML}
+        </div>
+        <div class="v2-footer-form-section v2-form">
+            <div class="form v2-footer-form-header">
+              ${fourthHeader?.outerHTML}
+            </div>
+            <div class="form footer-form">
+              ${formSection?.outerHTML}
+            </div>
+          </div>
+      </div>
+      <hr />
+      <div class="footer-bottom-section">
+        ${bottomLinksList?.outerHTML}
+      </div>
+    </div>
+  `;
+
+  const fragment = document.createRange().createContextualFragment(footerTemplate);
+  block.appendChild(fragment);
+  // block.append(footer);
+
+  // // make links to open in another browser tab/window
+  const socialLinks = block.querySelectorAll('.v2-footer-social-media-section a');
+  [...socialLinks].forEach((a) => { a.target = '_blank'; });
+
+  await decorateIcons(block);
+  await loadBlocks(block);
+
+  const targetNode = block.querySelector('.eloqua-form.block');
+  const observerOptions = {
+    childList: true,
+    attributes: false,
+    subtree: true,
+  };
+  let observer = null;
+  let submitButtonFixed = false;
+  let checkboxFixed = false;
+  const onFormLoaded = (mutationList) => {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const mutation of mutationList) {
+      if (submitButtonFixed && checkboxFixed) {
+        observer.disconnect();
+        return;
+      }
+
+      if (mutation.type === 'childList') {
+        const submitButton = block.querySelector('input[type="submit"]');
+        const emailInput = block.querySelector('input[name="emailAddress"]');
+        const label = emailInput.parentElement.querySelector('label');
+        const emailAndSubmitContainer = createElement('span', ['email-and-submit-container']);
+
+        // change the submit button to arrow button
+        // and display it sticked to the right side of email input
+        if (submitButton && emailInput) {
+          const parent = emailInput.parentElement;
+          submitButton.value = '→';
```

Comment:
Should be replaced with actual text which is visually hidden. The button should contain the SVG icon.

Reply:
Updated